File: src/main/java/cn/ua/goroddrev/domain/Warehouse.java
package cn.ua.goroddrev.domain;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "warehouses")
public class Warehouse {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "address")
    private String address;

    @OneToMany(mappedBy = "warehouse")
    private Set<WarehouseProductsInfo> warehouseProductsInfos;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Set<WarehouseProductsInfo> getWarehouseProductsInfos() {
        return warehouseProductsInfos;
    }

    public void setWarehouseProductsInfos(Set<WarehouseProductsInfo> warehouseProductsInfos) {
        this.warehouseProductsInfos = warehouseProductsInfos;
    }
}

File: src/main/webapp/gorod-drev-ui/src/app/shared/model/warehouse.ts
import {WarehouseProductInfo} from "./warehouse-product-info";

export class Warehouse {

  public constructor(private _id: number,
                     private _name: string,
                     private _address: string,
                     private _warehouseProductsInfos: WarehouseProductInfo[]) {
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get address(): string {
    return this._address;
  }

  set address(value: string) {
    this._address = value;
  }

  get warehouseProductsInfos(): WarehouseProductInfo[] {
    return this._warehouseProductsInfos;
  }

  set warehouseProductsInfos(value: WarehouseProductInfo[]) {
    this._warehouseProductsInfos = value;
  }
}
